File: flutter_document_scanner_platform_interface/lib/src/filter_type.dart
// Copyright (c) 2021, Christian Betancourt
// https://github.com/criistian14
//
// Use of this source code is governed by an MIT-style
// license that can be found in the LICENSE file or at
// https://opensource.org/licenses/MIT.

/// Filter types
enum FilterType {
  /// Without filter
  natural,

  /// Gray scale filter
  gray,

  /// Threshold filter
  eco,
}

/// Extension with utilities to FilterType
extension FilterTypeExt on FilterType {
  /// Return value of the enum
  String get value {
    switch (this) {
      case FilterType.natural:
        return 'NATURAL_FILTER';

      case FilterType.gray:
        return 'GRAY_FILTER';

      case FilterType.eco:
        return 'ECO_FILTER';
    }
  }
}

File: flutter_document_scanner_android/android/src/main/kotlin/com/christian/flutterDocumentScanner/FilterType.kt
package com.christian.flutterDocumentScanner

enum class FilterType(val filterValue: String) {
    NATURAL("NATURAL_FILTER"),
    GRAY("GRAY_FILTER"),
    ECO("ECO_FILTER");

    companion object {
        fun from(filter: String): FilterType {
            return values().find { it.filterValue.equals(filter, ignoreCase = true) }
                ?: throw IllegalArgumentException("Unsupported filter type: $filter")
        }
    }
}
